tests/chainer_tests/functions_tests/normalization_tests/test_standardize.py

```diff
@@ -1,15 +1,51 @@
+import functools
+import unittest
+
 import numpy
 
 from chainer.functions.normalization._standardize import _standardize
 from chainer import testing
 
 
-@testing.parameterize(*(testing.product({
-    'ch_out': [1, 5],
-    'size': [10, 20],
-    'dtype': [numpy.float32, numpy.float16],
-    'eps': [1e-5, 1e-1],
-})))
+def _skip_if(cond, reason):
+    """Skip test if cond(self) is True"""
+    def decorator(impl):
+        @functools.wraps(impl)
+        def wrapper(self, *args, **kwargs):
+            if cond(self):
+                raise unittest.SkipTest(reason)
+            else:
+                impl(self, *args, **kwargs)
+        return wrapper
+    return decorator
+
+
+def _is_good_param(param):
+    # Check if 'nonzero' param is valid and meaningful. On the latter point,
+    # x should contain at least a zero if 'nonzeros' param is given.
+    return param['nonzeros'] is None \
+        or param['nonzeros'] < numpy.prod(param['shape'])
+
+
+@testing.parameterize(*filter(
+    _is_good_param,
+    testing.product({
+        'ch_out': [1, 5],
+        'size': [10, 20],
+        'dtype': [numpy.float32, numpy.float16],
+    })
+    + testing.product([
+        # nonzeros (optional int): max number of nonzero elems in input
```

Comment:
Could you test with arrays many of whose elements are the same value, which is not necessarily zero?  For standardization, small std has to be tested, whereas `F.normalize` is tested with small norm.

Reply:
OK, now I do understand.
For test of `standardization`, valid test cases is that *all* of the elements are same (or slightly different) value, don't we?

Reply:
The new test param, (maybe `nonconsts`) would run over
- `None`: corresponds to the generic cases
- `0`: corresponds to the cases where all of the elements are the same
- `2`: corresponds to in-between cases